common/src/test/java/com/sequenceiq/cloudbreak/service/RetryTest.java

```diff
@@ -0,0 +1,79 @@
+package com.sequenceiq.cloudbreak.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+class RetryTest {
+
+    private static final String MESSAGE = "message";
+
+    @Test
+    void actionFailedExceptionConstructorTestWhenDefault() {
+        Retry.ActionFailedException underTest = new Retry.ActionFailedException();
+
+        assertThat(underTest.getMessage()).isNull();
+        assertThat(underTest.getCause()).isNull();
+    }
+
+    @Test
+    void actionFailedExceptionConstructorTestWhenMessage() {
+        Retry.ActionFailedException underTest = new Retry.ActionFailedException(MESSAGE);
+
+        assertThat(underTest.getMessage()).isEqualTo(MESSAGE);
+        assertThat(underTest.getCause()).isNull();
+    }
+
+    @Test
+    void actionFailedExceptionConstructorTestWhenMessageAndCause() {
+        Throwable cause = new RuntimeException();
+
+        Retry.ActionFailedException underTest = new Retry.ActionFailedException(MESSAGE, cause);
+
+        assertThat(underTest.getMessage()).isEqualTo(MESSAGE);
+        assertThat(underTest.getCause()).isSameAs(cause);
+    }
+
+    @Test
+    void actionFailedExceptionConstructorTestWhenCause() {
+        Throwable cause = new RuntimeException(MESSAGE);
+
+        Retry.ActionFailedException underTest = new Retry.ActionFailedException(cause);
+
+        assertThat(underTest.getMessage()).isEqualTo("java.lang.RuntimeException: message");
+        assertThat(underTest.getCause()).isSameAs(cause);
+    }
+
+    @Test
+    void actionFailedExceptionConstructorTestWhenMessageAndCauseAndEnableSuppressionAndWritableStackTrace() {
+        Throwable cause = new RuntimeException();
+        Throwable suppressed = new RuntimeException();
+
+        Retry.ActionFailedException underTest = new Retry.ActionFailedException(MESSAGE, cause, false, false);
+        underTest.addSuppressed(suppressed);
+
+        assertThat(underTest.getMessage()).isEqualTo(MESSAGE);
+        assertThat(underTest.getCause()).isSameAs(cause);
+        assertThat(underTest.getSuppressed()).isEmpty();
```

Comment:
why?

Reply:
Because that is how the `enableSuppression` flag works. TBH I have never used this constructor variant, but the impl in `Throwable` says that setting `enableSuppression = false` will make all `addSuppressed()` calls ignored. So here I am checking if exception suppression indeed works as expected.

There is no real alternative way to check this; the `enableSuppression` flag is not even saved in some `private` field, it is only used to initialize another field that controls the behavior of `addSuppressed()`.